packages/wallet/frontend/src/pages/_app.tsx

```diff
@@ -13,6 +19,60 @@ const titilium = Titillium_Web({
 
 export default function App({ Component, pageProps }: AppPropsWithLayout) {
   const getLayout = Component.getLayout ?? ((page) => page)
+  const { toast } = useToast()
+
+  useEffect(() => {
+    let socket: Socket | null = null
+    // Connect to the Socket.IO server
+    socket = io(process.env.NEXT_PUBLIC_BACKEND_URL ?? '', {
+      withCredentials: true
+    })
+
+    // Event listeners
+    socket?.on('MONEY_RECEIVED', (data) => {
+      toast({
+        description: (
+          <p>
+            <MoneyBird className="mr-2 inline-flex h-8 w-8 items-center justify-center" />
+            You received some {data.assetCode} into account {data.name}.
+          </p>
+        ),
+        variant: 'success'
+      })
+      updateBalance(
+        {
+          balance: data.balance,
+          assetCode: data.assetCode,
+          assetScale: data.assetScale
+        },
+        data.balance
+      )
+    })
+
+    socket?.on('MONEY_SENT', (data) => {
+      updateBalance(
+        {
+          balance: data.balance,
+          assetCode: data.assetCode,
+          assetScale: data.assetScale
+        },
+        data.balance
+      )
+    })
+
+    // log out the user when connection error occurs
+    socket.on('connect_error', async () => {
+      setTimeout(() => {
+        socket?.connect()
+      }, 1000)
+    })
+
```

Comment:
This would be good to have on rafiki.money as well:

```suggestion
socket.on('reconnect_error', (error) => {
  console.error('Reconnection error:', error);
});
```